Casks/d/dolphin@dev.rb

```diff
@@ -8,23 +8,24 @@
   homepage "https://dolphin-emu.org/"
 
   livecheck do
-    url "https://dolphin-emu.org/download/"
-    regex(%r{href=.*?/builds/([^/]+?)/([^/]+?)/dolphin[._-]master[._-]v?(\d+[a-z]?-\d+)-universal\.dmg}i)
-    strategy :page_match do |page, regex|
-      page.scan(regex).map { |match| "#{match[2]},#{match[0]},#{match[1]}" }
+    url "https://dolphin-emu.org/update/latest/dev/"
+    regex(%r{/builds/([^/]+?)/([^/]+?)/dolphin[._-]master[._-]v?(\d+[a-z]?-\d+)-universal\.dmg}i)
+    strategy :json do |json|
+      macos_artifact = json["artifacts"].find { |artifact| artifact["system"].include?("macOS") }
+      match = macos_artifact["url"].match(regex)
+      "#{match[3]},#{match[1]},#{match[2]}"
     end
```

Comment:
```suggestion
    regex(%r{/builds/([^/]+?)/([^/]+?)/dolphin.*?\.dmg}i)
    strategy :json do |json, regex|
      json["artifacts"]&.map do |artifact|
        match = artifact["url"]&.match(regex)
        next if match.blank?

        "#{json["shortrev"]},#{match[1]},#{match[2]}"
      end
    end
```

We can simplify the `regex` if we extract the version from `json["shortrev"]`. It might also be redundant to filter by the `system` property, since we are already matching the file extension.

Reply:
I changed it slightly such that it checks `system` for the correct URL with a simpler string comparison, then applies a single regex call to extract the characters from that URL.